stratum/hal/lib/barefoot/bfrt_node.cc

```diff
@@ -420,13 +408,25 @@ ::util::Status BfrtNode::WriteExternEntry(
     std::shared_ptr<BfSdeInterface::SessionInterface> session,
     const ::p4::v1::Update::Type type, const ::p4::v1::ExternEntry& entry) {
   switch (entry.extern_type_id()) {
-    case kTnaExternActionProfileId:
-    case kTnaExternActionSelectorId:
-      return bfrt_action_profile_manager_->WriteActionProfileEntry(session,
-                                                                   type, entry);
+    case kTnaExternActionProfileId: {
+      ::p4::v1::ActionProfileGroup act_prof_group;
+      CHECK_RETURN_IF_FALSE(entry.entry().UnpackTo(&act_prof_group))
+          << "Entry " << entry.ShortDebugString()
+          << " is not an action profile group.";
+      return bfrt_table_manager_->WriteActionProfileGroup(session, type,
+                                                          act_prof_group);
```

Comment:
@Yi-Tseng any reason no to unpack the extern here and then just call the regular `WriteActionProfileGroup` API?

Reply:
I forgot why I unpack it in the manager, but I agree that we can just unpack it here.

Reply:
By the way, the original logic is to unpack action profile member if it is an ActionProfile and unpack action profile group if it is an ActionSelector

Reply:
good catch! I swapped it back.